src/handshake/client.rs

```diff
@@ -52,15 +60,15 @@ impl<S: Read + Write> ClientHandshake<S> {
         let key = generate_key();
 
         let machine = {
-            let req = generate_request(request, &key)?;
+            let req = generate_request(request, &key, &mut config)?;
             HandshakeMachine::start_write(stream, req)
         };
 
         let client = {
             let accept_key = convert_key(key.as_ref()).unwrap();
             ClientHandshake {
                 verify_data: VerifyData { accept_key },
-                config,
+                config: Some(config),
```

Comment:
It looks a big counter-intuitive to me: why do we make it `Some(...)` if it's stored as an `Option`? Something is wrong with it. We either should not pass `Some` or not store it as `Option`? Or am I missing something?

Reply:
Upon the handshake finishing, the contents of the option are removed and the `WebSocket` instance is created. With some additional work, this could possibly be tidied up. Or, if the functionality to provide custom implementations of extensions is removed, then this will be removed.